worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -285,26 +298,28 @@ function onAfterProcessDocument(e) {
 
 function traverseDocumentForFailures(document) {
     processFailures(document);
-    for each(var subdoc in document.getSubdocuments()) {
-        traverseDocumentForFailures(subdoc);
+    if (document.getSubdocuments()) {
```

Comment:
Why the `if` here?
[Document::getSubdocuments()](https://cafdataprocessing.github.io/data-processing-service/pages/en-us/Creating_an_Action/apidocs/com/hpe/caf/worker/document/model/Document.html#getSubdocuments--) never returns `null`.